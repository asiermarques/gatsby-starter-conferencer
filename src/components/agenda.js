import React from "react"
import agendaMapper from "../lib/agendaMapper"
import agendaSlotType from "../lib/agendaSlotTypes";

export default ({agenda, speakers}) => {
    const tracks = agenda.tracks;
    const agendaSlotMap = agendaMapper(agenda, speakers);
    
    return <section id="agenda">
        <div className="container">
            <h2 className="text-center">Agenda</h2>
            <div className="table-responsive-sm">
            <table id="agenda-table" className="table table-striped stacktable large-only">
                    <thead>
                    <tr>
                        <th></th>
                        {tracks.map( (track, index) => <th key={index} width="40%">{track.name}</th> )}
                    </tr>
                    </thead>
                    <tbody>
                    {agendaSlotMap.map( (slot, slot_index) => { 
                        const [time_range, ...content] = slot;
                        return (<tr key={slot_index}>
                            <td class="time">{time_range}</td>
                            {content.map( (content_item, index) => {
                                
                                if(content_item.type === agendaSlotType.SPEAKER) { 
                                    return (
                                        <td key={index}>
                                            <a href={content_item.content.slug + "#talk"}>
                                                <h5>{content_item.content.talk.title}</h5>
                                                <p>{content_item.content.name}</p>
                                            </a>
                                        </td>
                                    )
                                }else{
                                    return (<td key={index}><p>{content_item.content}</p></td>)
                                }
                            })}
                        </tr>
                        )
                    })}
                    </tbody>
            </table>
            </div>
        </div>
    </section>
}

