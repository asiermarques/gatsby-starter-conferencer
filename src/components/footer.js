import React from "react"
import { graphql, StaticQuery } from "gatsby"
import LocationMap from "./locationMap"
import Sponsors from "./sponsors"
import Organizers from "./organizers"
import Nav from "./nav"


export default () => (
    <StaticQuery
        query={graphql`
        query {
            site {
              siteMetadata {
                conference_hashtag
                organizers {
                  name,
                  link,
                  image
                },
                footer_links {
                  link,
                  title
                },
                location {
                  gmaps_iframe_url,
                  addresses {
                    name,
                    line,
                    map_link
                  }
                },
                sponsor_blocks{
                  name,
                  height_em,
                  sponsors {
                      name,
                      link,
                      image
                  }
                }
              }
            }
          }
        `}
        render={ data => {

          const configData = data.site.siteMetadata;

          return (
            <footer>
                <LocationMap location={configData.location}/>
        
                <section id="about">
        
                  <div class="container">
        
                      <div class="row">
                            <div class="col-md-7">
                              <Organizers organizers={configData.organizers}/>
                            </div>
        
                            <div class="col-md-5 text-center">
                              <h4>About</h4>
                              <Nav items={configData.footer_links}></Nav>
                            </div>
        
                      </div>
                  </div>
        
                </section>
        
                <Sponsors sponsorBlocks={configData.sponsor_blocks}/>
        
                <section>
                    <div className="container">
                        <hr></hr>
                        Created with the <a href="https://github.com/asiermarques/gatsby-starter-conferencer">Gatsby Conferencer starter</a>
                    </div>
                </section>
        
                        
        
            </footer>
                )


        }}
    />
)