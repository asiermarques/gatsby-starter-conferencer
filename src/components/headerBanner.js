import React from "react"
import { graphql, StaticQuery } from "gatsby"


export default () => (
    <StaticQuery
        query={graphql`
        query {
            site {
                siteMetadata {
                    conference_name, 
                    conference_claim,
                    header_banner {
                        cta_pre_text,
                        cta_text
                    }
                }
            }
        }
        `}
        render={ data => (
            <div class="jumbotron">
                <div class="container">
                    <h1 class="display-4">{data.site.siteMetadata.conference_name}</h1>
                    <p class="lead">{data.site.siteMetadata.conference_claim}</p>
                    <hr class="my-4"/>
                    <p>{data.site.siteMetadata.header_banner.cta_pre_text}</p>
                    <a class="btn btn-primary btn-lg" href="/dot" role="button">{data.site.siteMetadata.header_banner.cta_text}</a>
                </div>
            </div>
        )}
    />
)