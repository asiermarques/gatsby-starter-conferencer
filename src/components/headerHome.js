import React from "react"

export default ({ data }) => <div class="jumbotron">
        <div class="container text-center">
            <h1 class="display-4">{data.conference_name}</h1>
            <p class="lead">{data.conference_claim}</p>
            <hr class="my-4"/>
            <p>{data.header_banner.cta_pre_text}</p>
            <a class="btn btn-primary btn-lg" href={data.header_banner.cta_url} role="button">{data.header_banner.cta_text}</a>
        </div>
    </div>