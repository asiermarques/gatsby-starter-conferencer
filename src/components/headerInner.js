import React from "react"

export default ({ data }) => <div class="jumbotron">
        <div class="container">
            <div className="row">
                <div className="col-sm-7">
                    <a href="/"><h2 class="display-4">{data.conference_name}</h2></a>
                    <p class="lead">{data.conference_claim}</p>
                </div>
                <div className="col-sm-5 text-center">
                    <p>{data.header_banner.cta_pre_text}</p>
                    <a class="btn btn-primary btn-lg" href={data.header_banner.cta_url} role="button">{data.header_banner.cta_text}</a>
                </div>
            </div> 
        </div>
    </div>
  

