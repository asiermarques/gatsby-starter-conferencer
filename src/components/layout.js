import React from "react"
// eslint-disable-next-line
import Styles from "../scss/custom.scss"

export default ({ children }) => (
  <div>
    {children}
  </div>
)