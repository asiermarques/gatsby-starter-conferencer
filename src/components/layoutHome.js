import React from "react"
import Styles from "../scss/custom.scss"

import Header from "./headerBanner"

export default ({ children }) => (
  <div>
    <Header></Header>
    {children}
  </div>
)

