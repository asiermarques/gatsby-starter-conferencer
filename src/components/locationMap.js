import React from "react"

export default ({ location }) => <section id="where">
        <h2 class="text-center">Location</h2>
        <div class="content_wrap">

          <div class="content">
            <div class="container">
              <div class="col-md-5 col-xs-12">

                <div class="card">
                  <div class="card-body">
                  {location.addresses.map( (address, index) => (<div key={index} class="address">

                    <h5>{address.name}</h5>
                    <p> 
                      <span dangerouslySetInnerHTML={{__html:address.line}}></span>
                      <br></br>
                      <small><a target="_blank" rel="noopener noreferrer" href={address.map_link}>open map</a></small>
                    </p>

                  </div>) )}
                  </div>
                </div>

              </div>
            </div>
          </div>

          <iframe src={location.gmaps_iframe_url}
          width="100%" 
          height="550" 
          title="map"
          frameborder="0" 
          allowfullscreen="false"></iframe>
        </div>
    </section>