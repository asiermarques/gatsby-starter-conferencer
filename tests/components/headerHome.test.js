import React from "react"
import renderer from "react-test-renderer"
import {PureHeader as Header} from "../../src/components/headerHome"
import HeaderConfigFixture from "../__fixtures/headerConfig"

describe("HeaderHome", () => {
  it("renders correctly", () => {
    const data = {
        site: {
          siteMetadata: HeaderConfigFixture,
        },
      }
    const tree = renderer.create(<Header  data={data}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})