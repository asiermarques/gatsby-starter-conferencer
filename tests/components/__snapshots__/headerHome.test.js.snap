// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderHome renders correctly 1`] = `
<div
  class="jumbotron"
>
  <div
    class="container text-center"
  >
    <h1
      class="display-4"
    >
      Our awesome conference
    </h1>
    <p
      class="lead"
    >
      A great conference for developers and other mystical creatures
    </p>
    <hr
      class="my-4"
    />
    <p>
      100 early-bird tickets left
    </p>
    <a
      class="btn btn-primary btn-lg"
      href="https://awesome_url_for_your_ticketing_platform.com"
      role="button"
    >
      Tickets on sale!
    </a>
  </div>
</div>
`;
